polaris-core/src/main/java/org/apache/polaris/core/storage/PolarisStorageConfigurationInfo.java

```diff
@@ -61,50 +62,25 @@
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME)
 @JsonSubTypes({
-  @JsonSubTypes.Type(value = AwsStorageConfigurationInfo.class),
-  @JsonSubTypes.Type(value = AzureStorageConfigurationInfo.class),
-  @JsonSubTypes.Type(value = GcpStorageConfigurationInfo.class),
-  @JsonSubTypes.Type(value = FileStorageConfigurationInfo.class),
+  @JsonSubTypes.Type(value = ImmutableAwsStorageConfigurationInfo.class),
```

Comment:
We're changing the types in this hierarchy quite a bit.
Can you add a test class (or tests in the individual sub-type test classes) to ensure that the serialized JSON format is still the same?